dispatch/dispatch.go

```diff
@@ -0,0 +1,56 @@
+package dispatch
+
+import (
+	"github.com/spacemonkeygo/errors"
+
+	"polydawn.net/repeatr/def"
+	"polydawn.net/repeatr/executor"
+	"polydawn.net/repeatr/executor/null"
+	"polydawn.net/repeatr/input"
+	"polydawn.net/repeatr/input/dir"
+	"polydawn.net/repeatr/input/tar"
+	"polydawn.net/repeatr/output"
+)
+
+// TODO: This should not require a global string -> class map :|
+// Should attempt to reflect-find, trying main package name first.
+// Will make simpler to use extended transports, etc.
+
+func GetExecutor(desire string) *executor.Executor {
+	var executor executor.Executor
+
+	switch desire {
+	case "null":
+		executor = &null.Executor{}
+	default:
+		panic(errors.NotImplementedError.New("No such executor %s", desire))
```

Comment:
If returning nil doesn't seem fun here, I'd consider these more `ValidationError`: this function is presumably handling user input or config data since otherwise the caller would just use the reference they meant directly, and this error is reached when the caller asked for a name that doesn't make sense.  NotImplemented would make more sense if it was like `case "coolPlannedThing": panic(NotImplemented)`.